src/NuGet.Core/NuGet.Packaging/Signing/ISignPackageReader.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// A readonly package that can provide signatures and a sign manifest from a package.
+    /// </summary>
+    public interface ISignPackageReader : IDisposable
+    {
+        /// <summary>
+        /// Get all signatures used to sign a package.
+        /// </summary>
+        /// <remarks>Returns an empty list if the package is unsigned.</remarks>
+        Task<IReadOnlyList<Signature>> GetSignaturesAsync(CancellationToken token);
+
+        /// <summary>
+        /// Read the signed manifest from a package.
+        /// </summary>
+        Task<SignManifest> GetSignManifestAsync(CancellationToken token);
```

Comment:
Why is there a requirement on the manifest to be signed? The signature would exist in a separate file. The manifest would contain no information about the signature.

Reply:
The term `SignManifest` is used here to differentiate it from the nuspec manifest, and the manifest classes used in pack.